cmd/imucalibrator/main.go

```diff
@@ -0,0 +1,116 @@
+/*
+imucalibrator calibrates IIM42652's sensors.
+
+Usage:
+
+	imucalibrator [flags]
+
+The flags are:
+
+	--sensor
+		Sensor to calibrate. This is a required flag. Possible values are
+		'gyro' and 'accelerometer'
+	--dev-path
+		Path to the spi device. By default, this is '/dev/spidev0.0'
+	--clear-calibration
+		Clears existing calibration data from the sensor set by the sensor flag.
+
+imucalibrator takes a sample of imu sensor data, averages it, then programs the imu
+user register to use the calculated average as a bias.
+*/
+package main
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/streamingfast/imu-controller/device/iim42652"
+)
+
+var (
+	sensor           = flag.String("sensor", "", "select sensor to calibrate or clear. Required. Values: gyro,accelerometer.")
+	devicePath       = flag.String("dev-path", "/dev/spidev0.0", "the dev path of the spi device Default is /dev/spidev0.0")
+	clearCalibration = flag.Bool("clear-calibration", false, "clear existing calibration data from the IMU")
+)
+
+func calibrateGyro(imuDevice *iim42652.IIM42652) error {
+	bias, err := imuDevice.CalibrateGyro()
+	if err != nil {
+		return fmt.Errorf("calibrating gyro: %w", err)
+	}
+	fmt.Println("gyro calibration values:", bias)
+	return nil
+}
+
+func calibrateAccelerometer(imuDevice *iim42652.IIM42652) error {
+	bias, err := imuDevice.CalibrateAccelerometer()
+	if err != nil {
+		return fmt.Errorf("calibrating accelerometer: %w", err)
+	}
+	fmt.Println("accel calibration values:", bias)
+	return nil
+}
+
+func validateFlags() error {
+	if *sensor == "" {
+		return fmt.Errorf("sensor flag is required")
+	}
+	if *sensor != "gyro" && *sensor != "accelerometer" {
+		return fmt.Errorf("sensor '%v' not recognized, must be 'gyro' or 'accelerometer'", *sensor)
+	}
+	return nil
+}
+
+func main() {
+	flag.Parse()
+
+	if err := validateFlags(); err != nil {
+		panic(fmt.Errorf("validateflags: %w", err))
+	}
+
+	// Note: Only 16G and 2000dps work. iim42652 module does not
+	// configure these parameters on the device and relies on the defaults.
+	imuDevice := iim42652.NewSpi(
+		*devicePath,
+		iim42652.AccelerationSensitivityG16,
+		iim42652.GyroScalesG2000,
+		true,
+		false, // skip power management
+	)
+
+	err := imuDevice.Init()
+	if err != nil {
+		panic(fmt.Errorf("initializing IMU: %w", err))
+	}
+
+	if *sensor == "gyro" {
+		if *clearCalibration {
+			err := imuDevice.ClearGyroBias()
+			if err != nil {
+				panic(fmt.Errorf("clearing IMU: %w", err))
+			}
+			fmt.Println("Gyro cleared!")
+		} else {
+			err := calibrateGyro(imuDevice)
+			if err != nil {
+				panic(fmt.Errorf("calibrating IMU: %w", err))
```

Comment:
@mazar-farran Do you think it makes sense to do anything else here except panicking?
Maybe, clearing the calibration? Re-initialise the device?
Just thinking out loud